Riot/Managers/Settings/Shared/RiotSharedSettings.swift

```diff
@@ -52,6 +52,39 @@ class RiotSharedSettings: NSObject {
 
     // MARK: - Public
 
+    // MARK: Integration provisioning
+
+    func hasIntegrationProvisioningEnabled() -> Bool {
+        return getIntegrationProvisioning()?.enabled ?? true
+    }
+
+    func getIntegrationProvisioning() -> RiotSettingIntegrationProvisioning? {
+        guard let integrationProvisioningDict = getAccountData(forEventType: Settings.integrationProvisioning) else {
+            return nil
+        }
+
+        do {
+            let integrationProvisioning: RiotSettingIntegrationProvisioning = try serializationService.deserialize(integrationProvisioningDict)
+            return integrationProvisioning
+        } catch {
+            return nil
+        }
```

Comment:
In this case you can write:
`return try? serializationService.deserialize(integrationProvisioningDict)`